asllib/env.ml

```diff
@@ -30,7 +30,11 @@ module type RunTimeConf = sig
 end
 
 module RunTime (C : RunTimeConf) = struct
-  type func = int ref * AST.func
+  type func = {
+    call_id : int ref;
+    static_def : AST.func;
+    fuel : int ref option;
```

Comment:
I'm not sure it's a good idea to have a reference here. It's probably better to set the fuel to the recursion limit every time a call is made from a non-recursive subprogram.
Also, I don't see where you reset the fuel after a recursive call is done to enable calling it next time with the rull recursion limit.

Reply:
Let's discuss offline.